src/store/configureStore.ts

```diff
@@ -1,30 +1,43 @@
 import * as Sentry from '@sentry/browser'
-import {
-  routerMiddleware,
-  replace as replaceRoute
-} from 'connected-react-router'
+import { routerMiddleware, push as pushRoute } from 'connected-react-router'
 import { debounce, pick } from 'lodash'
-import { createStore, applyMiddleware } from 'redux'
+import { createStore, applyMiddleware, Action, Store } from 'redux'
 import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction'
 import createSagaMiddleware from 'redux-saga'
 import createSentryMiddleware from 'redux-sentry-middleware'
 
 import { reducers as clientStoreReducers } from 'common/store'
 import { postMessage } from 'services/native-mobile-interface/helpers'
 import { MessageType } from 'services/native-mobile-interface/types'
+import { track as amplitudeTrack } from 'store/analytics/providers/amplitude'
 import createRootReducer from 'store/reducers'
 import rootSaga from 'store/sagas'
-import { getIsDeployedOnHost } from 'utils/clientUtil'
 import history from 'utils/history'
+import logger from 'utils/logger'
 import { ERROR_PAGE } from 'utils/route'
 
+import { AppState } from './types'
+
+declare global {
+  interface Window {
+    store: any
+  }
+}
+
 const NATIVE_MOBILE = process.env.REACT_APP_NATIVE_MOBILE
 
-const onSagaError = (error, errorInfo) => {
+type RootState = ReturnType<typeof store.getState>
+
+const onSagaError = (
+  error: Error,
+  errorInfo: {
+    sagaStack: string
+  }
+) => {
   console.error(`Caught saga error: ${error} ${errorInfo}`)
-  store.dispatch(replaceRoute(ERROR_PAGE))
+  store.dispatch(pushRoute(ERROR_PAGE))
 
-  if (!getIsDeployedOnHost()) return
+  // Try recording to sentry
   try {
     Sentry.withScope(scope => {
```

Comment:
maybe we can try to reuse these blocks with the other error saga so the impl is consistent?

Reply:
good idea